### ISSUE TITLE
Too many open files in system

### ISSUE THREAD

**Issue by dpoltoratsky** [flosse/sloc#83]
When I try to run sloc on my application I get an exception:

```
# sloc app
(libuv) kqueue(): Too many open files in system
events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: write EBADF
    at exports._errnoException (util.js:1026:11)
    at WriteStream.Socket._writeGeneric (net.js:715:26)
    at WriteStream.Socket._write (net.js:734:8)
    at doWrite (_stream_writable.js:333:12)
    at writeOrBuffer (_stream_writable.js:319:5)
    at WriteStream.Writable.write (_stream_writable.js:246:11)
    at WriteStream.Socket.write (net.js:661:40)
    at Console.warn (console.js:51:16)
    at /usr/local/lib/node_modules/sloc/lib/cli.js:206:22
    at /usr/local/lib/node_modules/sloc/node_modules/async/lib/async.js:52:16
```

**Comment by flosse** [comment]
Thanks for reporting that issue.

**Comment by dpoltoratsky** [comment]
No problem. Just and addition, I ended up running the tool against individual subfolders and the total number of lines in application turned out to be about one million. Hope this helps.

**Comment by natew** [comment]
Getting this as well

```
(libuv) kqueue(): Too many open files
events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: write EBADF
    at exports._errnoException (util.js:1036:11)
    at WriteStream.Socket._writeGeneric (net.js:705:26)
    at WriteStream.Socket._write (net.js:724:8)
    at doWrite (_stream_writable.js:307:12)
    at writeOrBuffer (_stream_writable.js:293:5)
    at WriteStream.Writable.write (_stream_writable.js:220:11)
    at WriteStream.Socket.write (net.js:651:40)
    at Console.warn (console.js:51:16)
    at /Users/nw/n/lib/node_modules/sloc/lib/cli.js:206:22
    at /Users/nw/n/lib/node_modules/sloc/node_modules/async/lib/async.js:52:16
```

**Comment by flosse** [comment]
How can this be reproduced and tested? It would be helpful to have a test case that covers this.
Any ideas?

**Comment by tomByrer** [comment]
1,000,000 LOC is either many files or several big files; not sure if those should be stored in this repo.  (Reminds me of when I had to download the jsDelivr CDN... nightmare)

Could make a program that creates a bunch of fake files, but both the creation & processing those files would take time & HD space.  Maybe an optional test?

**Comment by dpoltoratsky** [comment]
In my case, it was a combination of a big number of files with some quite big files.

As to testing, I would create a simple test that generates a lot files with some code in them.

Below is simplistic example of how I would do it in PHP:
```php
for ($i=0; $i<100000; $i++) {
    $fp = fopen(microtime(true), 'w');
    fwrite($fp, 'put like a 100 of lines of code here...');
    fclose($fp);
}
```

Hope this helps!

**Comment by assembledadam** [comment]
Got this too.

**Comment by ratheeshr77** [comment]
Works if I use async.mapLimit() instead of async.map() in readDir() in cli.js.

Eg. async.mapLimit(filterFiles(res.files), 1000, processFile, done);

I had about 4500 files.

**Comment by miclaus** [comment]
+1 

I have a lot of `node_modules`.

**Comment by flosse** [comment]
Would some create a PR?
thx

**Comment by ungoldman** [comment]
PR opened using @ratheeshr77's fix above

**Comment by ungoldman** [comment]
Thanks for merging @flosse! Any chance of this fix landing in a release soon?

**Comment by flosse** [comment]
I just released v0.2.0 :)

**Comment by ungoldman** [comment]
yay, thanks!
